dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,551 @@
+            return Result.NEXT;
+        }
+
+        try {
+
+            if (null != this.samlConfig() && null != session && this.isAnySamlConfigurated()) {
+
+                final Host host = hostWebAPI.getCurrentHostNoThrow(request);
+                final IdentityProviderConfiguration identityProviderConfiguration = // gets the SAML Configuration for this site.
+                        this.identityProviderConfigurationFactory.findIdentityProviderConfigurationById(
+                                host.getIdentifier());
+
+                // If idpConfig is null, means this site does not need SAML processing
+                if (null != identityProviderConfiguration && identityProviderConfiguration.isEnabled()) { // SAML is configurated, so continue
+
+                    // check if there is any exception filter path, to avoid to canApply all the logic.
+                    if (!this.checkAccessFilters(request.getRequestURI(), this.getAccessFilterArray(identityProviderConfiguration))
+                            && this.checkIncludePath(request.getRequestURI(), this.getIncludePathArray(identityProviderConfiguration))) {
+
+                        if (this.samlWebUtils.isNotLogged(request)) {
+
+                            final AutoLoginResult autoLoginResult = this.autoLogin(request, response, session, identityProviderConfiguration);
+
+                            // we have to assign again the session, since the doAutoLogin might be renewed.
+                            session = autoLoginResult.getSession();
+
+                            // if the auto login couldn't logged the user, then send it to the IdP login page (if it is not already logged in).
+                            if (null == session || !autoLoginResult.isAutoLogin() || this.samlWebUtils.isNotLogged(request)) {
+
+                                this.doAuthentication(request, response, session, identityProviderConfiguration);
+                                return Result.SKIP_NO_CHAIN;
+                            }
+                        }
+                    }
+
+                    final boolean isLogoutNeed = this.samlConfigurationService.getConfigAsBoolean(
+                            identityProviderConfiguration, SamlName.DOTCMS_SAML_IS_LOGOUT_NEED);
+                    // Starting the logout if it is logout
+                    Logger.debug(this, ()-> "----------------------------- doFilter --------------------------------");
+                    Logger.debug(this, ()-> "- isLogoutNeed = " + isLogoutNeed);
+                    Logger.debug(this, ()-> "- httpServletRequest.getRequestURI() = " + request.getRequestURI());
+
+                    if (isLogoutNeed &&
+                            this.samlWebUtils.isLogoutRequest(request.getRequestURI(), this.getLogoutPathArray(identityProviderConfiguration))) {
+
+                        if (this.doLogout(response, request, session, identityProviderConfiguration)) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)